tests/cecil-tests/AttributeTest.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics;
+
+using NUnit.Framework;
+
+using Mono.Cecil;
+
+#nullable enable
+
+namespace Cecil.Tests {
+
+	// [TestFixture]
+	public class AttributeTest {
+		// https://github.com/xamarin/xamarin-macios/issues/10170
```

Comment:
Another test scenario:

Every API must either have no attributes at all, or have an availability attribute for the current platform (i.e. every type (or method or property or ...) in Xamarin.iOS.dll must have a `[SupportedOSPlatform ("ios*")]` or no availability attributes at all).

Except: enum fields, for enums we often (on purpose) include fields that only exists on other platforms.